pkg/operator/loglevel/logging_controller.go

```diff
@@ -51,23 +51,20 @@ func (c LogLevelController) sync() error {
 		return err
 	}
 
-	logLevel := fmt.Sprintf("%d", LogLevelToKlog(detailedSpec.OperatorLogLevel))
+	currentLogLevel := CurrentLogLevel()
 
-	var level klog.Level
-
-	oldLevel, ok := level.Get().(klog.Level)
-	if !ok {
-		oldLevel = level
+	// When the current loglevel is the desired one, do nothing
+	if currentLogLevel == detailedSpec.OperatorLogLevel {
+		return nil
 	}
 
-	if err := level.Set(logLevel); err != nil {
+	// Set the new loglevel if the operator spec changed
+	if err := SetVerbosityFlagValue(detailedSpec.OperatorLogLevel); err != nil {
 		c.eventRecorder.Warningf("LoglevelChangeFailed", "Unable to set loglevel level %v", err)
```

Comment:
Why we're such fans of repeating the level word :wink:

Reply:
i fixed these